packages/wallets/src/components/WalletsResetMT5Password/WalletsResetMT5Password.tsx

```diff
@@ -70,25 +72,29 @@ const WalletsResetMT5Password = ({
 
     useEffect(() => {
         if (isChangePasswordSuccess) {
-            localStorage.removeItem(`verification_code.${actionParams}`); // TODO:Remove verification code from local storage
             show(<WalletSuccessResetMT5Password title={title} />, { defaultRootId: 'wallets_modal_root' });
         } else if (isChangePasswordError) {
             hide();
         }
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [platform, title, actionParams, isChangePasswordSuccess, isChangePasswordError]);
+    }, [platform, title, isChangePasswordSuccess, isChangePasswordError]);
 
     useEffect(() => {
         if (isChangeInvestorPasswordSuccess) {
-            localStorage.removeItem(`verification_code.${actionParams}`); // TODO:Remove verification code from local storage
             show(<WalletSuccessResetMT5Password isInvestorPassword title={title} />, {
                 defaultRootId: 'wallets_modal_root',
             });
         } else if (isChangeInvestorPasswordError) {
             hide();
         }
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [platform, title, actionParams, isChangeInvestorPasswordSuccess, isChangeInvestorPasswordError]);
+    }, [platform, title, isChangeInvestorPasswordSuccess, isChangeInvestorPasswordError]);
+
+    useEffect(() => {
+        if (changeInvestorPasswordStatus || changePasswordStatus) {
+            localStorage.removeItem(`verification_code.${actionParams}`);
```

Comment:
how urgent this fix is? would it make sense to actually remove the use of local storage for storing verification_code and just use state for it? 

this type of stuff shouldn't be put to local storage to begin with, as it opens an whole rabbit hole of engineering and security issues.

Reply:
if its not urgent - probably worth spending more time on proper implementation,

Reply:
It is @wojciech-deriv. I do agree with you, there is a better way of handling it. However, there many dependent code within the stores from core. I have a discussion with the tradershub team they are using cookies and session storage for this. Maybe in a improvment card we can also use this approach aswell. What do you think?
cc: @nijil-deriv

Reply:
for verification code, we should use useState or pass as props (basically handle within app itself),

Reason being, that verification tokens are generally short-lived, or single use beings, so no reason to store them in any persistent place like cookies / session / local storage. It generates stability and secuiryt risks to store short -lived tokens in any persistent database. Espeically cookies which tends to be send with every request to backend (unless they are set in a very specific way)

Reply:
example on how ti could be done without local storage: 
/deriv-app/packages/wallets/src/features/cashier/flows/WalletWithdrawal/WalletWithdrawal.tsx

Reply:
If so can we do it in a seperate task because it is blocking some of the other subtasks for me. Also the story point estimation should be recalculated don't you think so?